test/core/parser/segments/segments_base_test.py

```diff
@@ -228,35 +210,12 @@ def test__parser__base_segments_base_compare():
     assert ds1 != dsa2
 
 
-def test__parser__base_segments_file(raw_seg_list):
-    """Test BaseFileSegment to behave as expected."""
-    base_seg = BaseFileSegment(raw_seg_list, fname="/some/dir/file.sql")
-    assert base_seg.type == "file"
-    assert base_seg.file_path == "/some/dir/file.sql"
-    assert base_seg.can_start_end_non_code
-    assert base_seg.allow_empty
-
-
-def test__parser__raw_get_raw_segments(raw_seg_list):
-    """Test niche case of calling get_raw_segments on a raw segment."""
-    for s in raw_seg_list:
-        assert s.get_raw_segments() == [s]
-
-
-def test__parser__raw_segments_with_ancestors(raw_seg_list):
-    """Test raw_segments_with_ancestors.
-
-    This is used in the reflow module to assess parse depth.
-    """
-    test_seg = DummySegment([DummyAuxSegment(raw_seg_list[:1]), raw_seg_list[1]])
-    # Result should be the same raw segment, but with appropriate parents
-    assert test_seg.raw_segments_with_ancestors == [
-        (
-            raw_seg_list[0],
-            [
-                PathStep(test_seg, 0, 2, (0, 1)),
-                PathStep(test_seg.segments[0], 0, 1, (0,)),
-            ],
-        ),
-        (raw_seg_list[1], [PathStep(test_seg, 1, 2, (0, 1))]),
-    ]
+def test__parser__base_segments_pickle_safe(raw_seg_list):
+    """Test pickling and unpickling of BaseSegment."""
+    test_seg = BaseSegment([BaseSegment(raw_seg_list)])
+    test_seg.populate_parents()
```

Comment:
i think populate_parents is confusing as a method name. It is populating the children's parents, but running this as a function on test_seg implies you're setting test_seg's parent. Not the end of the world, but maybe there's a more clear name

Reply:
Hmmm that is a good point 🤔 . I'm going to see if I can come up with a better name...

Reply:
How about `set_as_parent()`? That feels more descriptive.

Reply:
Yeah, I prefer that